From 605f6fa0efcc9f017cc7cfaee2ec7956004941fb Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <accounts+github@chiller3.com>
Date: Thu, 1 Aug 2024 11:07:08 -0400
Subject: [PATCH] sbctl: Fix human readable output being printed when using
 --json

`PersistentPreRun` was being overwritten later in `main()`, causing
`logging.PrintOff()` to never be called.

Signed-off-by: Andrew Gunnerson <accounts+github@chiller3.com>
---
 cmd/sbctl/main.go | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/cmd/sbctl/main.go b/cmd/sbctl/main.go
index 72a7b09..24f6f83 100644
--- a/cmd/sbctl/main.go
+++ b/cmd/sbctl/main.go
@@ -64,15 +64,6 @@ func baseFlags(cmd *cobra.Command) {
 	flags.BoolVar(&cmdOptions.DisableLandlock, "disable-landlock", false, "Disable landlock sandboxing")
 	flags.BoolVar(&cmdOptions.Debug, "debug", false, "Enable verbose debug logging")
 	flags.StringVarP(&cmdOptions.Config, "config", "", "", "Path to configuration file")
-
-	cmd.PersistentPreRun = func(cmd *cobra.Command, args []string) {
-		if cmdOptions.JsonOutput {
-			logging.PrintOff()
-		}
-		if cmdOptions.QuietOutput {
-			logging.DisableInfo = true
-		}
-	}
 }
 
 func JsonOut(v interface{}) error {
@@ -134,6 +125,12 @@ func main() {
 
 	// We need to set this after we have parsed stuff
 	rootCmd.PersistentPreRun = func(_ *cobra.Command, _ []string) {
+		if cmdOptions.JsonOutput {
+			logging.PrintOff()
+		}
+		if cmdOptions.QuietOutput {
+			logging.DisableInfo = true
+		}
 		if cmdOptions.DisableLandlock {
 			state.Config.Landlock = false
 		}
